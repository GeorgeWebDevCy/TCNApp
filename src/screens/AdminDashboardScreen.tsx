import React, { useCallback, useEffect, useMemo, useState } from 'react';
import {
  ActivityIndicator,
  Alert,
  RefreshControl,
  ScrollView,
  StyleSheet,
  Text,
  View,
  Pressable,
} from 'react-native';
import { SafeAreaView } from 'react-native-safe-area-context';
import { useAuthContext } from '../contexts/AuthContext';
import { useLocalization } from '../contexts/LocalizationContext';
import {
  AdminAccountSummary,
  approveVendorAccount,
  fetchAdminAccounts,
  rejectVendorAccount,
} from '../services/adminService';
import { COLORS } from '../config/theme';
import { useResponsiveLayout } from '../hooks/useResponsiveLayout';

interface AdminDashboardScreenProps {
  onOpenMemberExperience?: () => void;
}

type ActiveAction = {
  id: number;
  type: 'approve' | 'reject';
} | null;

const resolveStatusLabel = (
  t: ReturnType<typeof useLocalization>['t'],
  status: string | null,
) => {
  if (!status) {
    return t('admin.dashboard.statusLabels.pending');
  }

  const normalized = status.toLowerCase();
  const key = `admin.dashboard.statusLabels.${normalized}`;
  return t(key, {
    defaultValue: t('admin.dashboard.statusLabels.pending'),
  });
};

const resolveStatusStyle = (status: string | null) => {
  const normalized = status ? status.toLowerCase() : 'pending';
  if (normalized === 'active') {
    return styles.statusPillActive;
  }
  if (normalized === 'rejected') {
    return styles.statusPillRejected;
  }
  if (normalized === 'suspended') {
    return styles.statusPillSuspended;
  }
  return styles.statusPillPending;
};

export const AdminDashboardScreen: React.FC<AdminDashboardScreenProps> = ({
  onOpenMemberExperience,
}) => {
  const { state, getSessionToken } = useAuthContext();
  const { t } = useLocalization();
  const layout = useResponsiveLayout();

  const [accounts, setAccounts] = useState<AdminAccountSummary[]>([]);
  const [loading, setLoading] = useState(true);
  const [refreshing, setRefreshing] = useState(false);
  const [error, setError] = useState<string | null>(null);
  const [activeAction, setActiveAction] = useState<ActiveAction>(null);

  const loadAccounts = useCallback(async () => {
    setLoading(true);
    setError(null);
    try {
      const token = await getSessionToken();
      if (!token) {
        throw new Error(t('admin.dashboard.errors.load'));
      }
      const directory = await fetchAdminAccounts(token);
      setAccounts(directory);
    } catch (err) {
      const message =
        err instanceof Error ? err.message : t('admin.dashboard.errors.load');
      setError(message);
    } finally {
      setLoading(false);
    }
  }, [getSessionToken, t]);

  useEffect(() => {
    void loadAccounts();
  }, [loadAccounts]);

  const onRefresh = useCallback(async () => {
    setRefreshing(true);
    try {
      await loadAccounts();
    } finally {
      setRefreshing(false);
    }
  }, [loadAccounts]);

  const pendingVendors = useMemo(
    () =>
      accounts.filter(account => {
        const isVendor = account.accountType === 'vendor';
        const status = account.vendorStatus ?? account.accountStatus;
        return isVendor && (status ?? 'pending').toLowerCase() === 'pending';
      }),
    [accounts],
  );

  const handleActionResult = useCallback(
    async (
      account: AdminAccountSummary,
      action: 'approve' | 'reject',
      performer: (token: string) => Promise<void>,
      errorKey: 'approve' | 'reject',
    ) => {
      setActiveAction({ id: account.id, type: action });
      try {
        const token = await getSessionToken();
        if (!token) {
          throw new Error(t(`admin.dashboard.errors.${errorKey}`));
        }
        await performer(token);
        await loadAccounts();
      } catch (err) {
        const message =
          err instanceof Error
            ? err.message
            : t(`admin.dashboard.errors.${errorKey}`);
        Alert.alert(t('admin.dashboard.title'), message);
      } finally {
        setActiveAction(null);
      }
    },
    [getSessionToken, loadAccounts, t],
  );

  const confirmApprove = useCallback(
    (account: AdminAccountSummary) => {
      const displayName = account.name || account.email;
      Alert.alert(
        t('admin.dashboard.confirmations.approveTitle'),
        t('admin.dashboard.confirmations.approveMessage', {
          replace: { name: displayName },
        }),
        [
          {
            text: t('admin.dashboard.actions.cancel'),
            style: 'cancel',
          },
          {
            text: t('admin.dashboard.actions.approve'),
            style: 'default',
            onPress: () =>
              void handleActionResult(
                account,
                'approve',
                token => approveVendorAccount(token, account.id),
                'approve',
              ),
          },
        ],
      );
    },
    [handleActionResult, t],
  );

  const confirmReject = useCallback(
    (account: AdminAccountSummary) => {
      const displayName = account.name || account.email;
      Alert.alert(
        t('admin.dashboard.confirmations.rejectTitle'),
        t('admin.dashboard.confirmations.rejectMessage', {
          replace: { name: displayName },
        }),
        [
          {
            text: t('admin.dashboard.actions.cancel'),
            style: 'cancel',
          },
          {
            text: t('admin.dashboard.actions.reject'),
            style: 'destructive',
            onPress: () =>
              void handleActionResult(
                account,
                'reject',
                token => rejectVendorAccount(token, account.id),
                'reject',
              ),
          },
        ],
      );
    },
    [handleActionResult, t],
  );

  const canReviewVendor = useCallback((account: AdminAccountSummary) => {
    if (account.accountType !== 'vendor') {
      return false;
    }
    const status = (account.vendorStatus ?? account.accountStatus ?? 'pending').toLowerCase();
    return status === 'pending';
  }, []);

  const isActionInFlight = useCallback(
    (accountId: number, type: 'approve' | 'reject') =>
      activeAction?.id === accountId && activeAction.type === type,
    [activeAction],
  );

  const pendingSummaryLabel = useMemo(() => {
    const count = pendingVendors.length;
    if (count === 0) {
      return t('admin.dashboard.pendingSummaryNone');
    }
    return t('admin.dashboard.pendingSummary', {
      replace: { count: count.toString() },
    });
  }, [pendingVendors.length, t]);

  const headerName = state.user?.name ?? state.user?.email ?? '';

  const refreshControl = (
    <RefreshControl
      refreshing={refreshing}
      onRefresh={() => {
        void onRefresh();
      }}
      tintColor={COLORS.primary}
    />
  );

  return (
    <SafeAreaView style={styles.safeArea}>
      <ScrollView
        contentContainerStyle={[styles.container, { padding: layout.contentPadding }]}
        refreshControl={refreshControl}
      >
        <View style={styles.header}>
          <View style={styles.headerText}>
            <Text style={styles.title}>{t('admin.dashboard.title')}</Text>
            <Text style={styles.subtitle}>{t('admin.dashboard.subtitle')}</Text>
            {headerName ? (
              <Text style={styles.operator}>{headerName}</Text>
            ) : null}
          </View>
          {onOpenMemberExperience ? (
            <Pressable
              onPress={onOpenMemberExperience}
              style={styles.secondaryButton}
              accessibilityRole="button"
            >
              <Text style={styles.secondaryButtonText}>
                {t('admin.dashboard.viewMemberApp')}
              </Text>
            </Pressable>
          ) : null}
        </View>

        <View style={styles.summaryCard}>
          <Text style={styles.summaryTitle}>{pendingSummaryLabel}</Text>
          <Text style={styles.summaryHint}>{t('admin.dashboard.refreshHint')}</Text>
        </View>

        {error ? <Text style={styles.errorText}>{error}</Text> : null}

        {loading ? (
          <View style={styles.loadingContainer}>
            <ActivityIndicator color={COLORS.primary} size="large" />
          </View>
        ) : (
          <View style={styles.sections}>
            <View style={styles.section}>
              <Text style={styles.sectionTitle}>
                {t('admin.dashboard.sections.pending')}
              </Text>
              {pendingVendors.length === 0 ? (
                <Text style={styles.emptyMessage}>
                  {t('admin.dashboard.empty.pending')}
                </Text>
              ) : (
                pendingVendors.map(account => {
                  const status = account.vendorStatus ?? account.accountStatus;
                  return (
                    <View style={styles.accountCard} key={`pending-${account.id}`}>
                      <View style={styles.accountHeader}>
                        <View style={styles.accountTitleBlock}>
                          <Text style={styles.accountName}>{account.name}</Text>
                          <Text style={styles.accountEmail}>{account.email}</Text>
                        </View>
                        <View
                          style={[styles.statusPill, resolveStatusStyle(status)]}
                        >
                          <Text style={styles.statusPillText}>
                            {resolveStatusLabel(t, status ?? null)}
                          </Text>
                        </View>
                      </View>
                      <Text style={styles.accountMeta}>
                        {t('admin.dashboard.tableHeaders.type')}: {account.accountType ?? 'â€”'}
                      </Text>
                      <View style={styles.actionsRow}>
                        <Pressable
                          style={[styles.primaryButton, styles.actionButton]}
                          onPress={() => confirmApprove(account)}
                          disabled={isActionInFlight(account.id, 'approve')}
                          accessibilityRole="button"
                        >
                          {isActionInFlight(account.id, 'approve') ? (
                            <ActivityIndicator
                              color={COLORS.textOnPrimary}
                              size="small"
                            />
                          ) : (
                            <Text style={styles.primaryButtonText}>
                              {t('admin.dashboard.actions.approve')}
                            </Text>
                          )}
                        </Pressable>
                        <Pressable
                          style={[styles.dangerButton, styles.actionButton]}
                          onPress={() => confirmReject(account)}
                          disabled={isActionInFlight(account.id, 'reject')}
                          accessibilityRole="button"
                        >
                          {isActionInFlight(account.id, 'reject') ? (
                            <ActivityIndicator color={COLORS.textOnPrimary} size="small" />
                          ) : (
                            <Text style={styles.dangerButtonText}>
                              {t('admin.dashboard.actions.reject')}
                            </Text>
                          )}
                        </Pressable>
                      </View>
                    </View>
                  );
                })
              )}
            </View>

            <View style={styles.section}>
              <Text style={styles.sectionTitle}>
                {t('admin.dashboard.sections.directory')}
              </Text>
              {accounts.length === 0 ? (
                <Text style={styles.emptyMessage}>
                  {t('admin.dashboard.empty.directory')}
                </Text>
              ) : (
                accounts.map(account => {
                  const status = account.accountStatus ?? account.vendorStatus;
                  const canReview = canReviewVendor(account);

                  return (
                    <View style={styles.accountCard} key={`account-${account.id}`}>
                      <View style={styles.accountHeader}>
                        <View style={styles.accountTitleBlock}>
                          <Text style={styles.accountName}>{account.name}</Text>
                          <Text style={styles.accountEmail}>{account.email}</Text>
                        </View>
                        <View
                          style={[styles.statusPill, resolveStatusStyle(status)]}
                        >
                          <Text style={styles.statusPillText}>
                            {resolveStatusLabel(t, status ?? null)}
                          </Text>
                        </View>
                      </View>
                      <View style={styles.accountMetaRow}>
                        <Text style={styles.accountMeta}>
                          {t('admin.dashboard.tableHeaders.type')}: {account.accountType ?? 'â€”'}
                        </Text>
                        {account.createdAt ? (
                          <Text style={styles.accountMeta}>
                            {account.createdAt}
                          </Text>
                        ) : null}
                      </View>
                      {canReview ? (
                        <View style={styles.actionsRow}>
                          <Pressable
                            style={[styles.primaryButton, styles.actionButton]}
                            onPress={() => confirmApprove(account)}
                            disabled={isActionInFlight(account.id, 'approve')}
                            accessibilityRole="button"
                          >
                            {isActionInFlight(account.id, 'approve') ? (
                              <ActivityIndicator
                                color={COLORS.textOnPrimary}
                                size="small"
                              />
                            ) : (
                              <Text style={styles.primaryButtonText}>
                                {t('admin.dashboard.actions.approve')}
                              </Text>
                            )}
                          </Pressable>
                          <Pressable
                            style={[styles.dangerButton, styles.actionButton]}
                            onPress={() => confirmReject(account)}
                            disabled={isActionInFlight(account.id, 'reject')}
                            accessibilityRole="button"
                          >
                            {isActionInFlight(account.id, 'reject') ? (
                              <ActivityIndicator
                                color={COLORS.textOnPrimary}
                                size="small"
                              />
                            ) : (
                              <Text style={styles.dangerButtonText}>
                                {t('admin.dashboard.actions.reject')}
                              </Text>
                            )}
                          </Pressable>
                        </View>
                      ) : null}
                    </View>
                  );
                })
              )}
            </View>
          </View>
        )}
      </ScrollView>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  safeArea: {
    flex: 1,
    backgroundColor: COLORS.background,
  },
  container: {
    flexGrow: 1,
    gap: 16,
  },
  header: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'flex-start',
    gap: 12,
  },
  headerText: {
    flex: 1,
    gap: 8,
  },
  title: {
    fontSize: 24,
    fontWeight: '700',
    color: COLORS.textPrimary,
  },
  subtitle: {
    fontSize: 14,
    color: COLORS.textSecondary,
  },
  operator: {
    fontSize: 12,
    color: COLORS.textSecondary,
  },
  secondaryButton: {
    backgroundColor: COLORS.surface,
    borderRadius: 12,
    paddingHorizontal: 16,
    paddingVertical: 10,
    borderWidth: 1,
    borderColor: COLORS.border,
  },
  secondaryButtonText: {
    color: COLORS.primary,
    fontWeight: '600',
  },
  summaryCard: {
    backgroundColor: COLORS.surface,
    borderRadius: 16,
    padding: 16,
    gap: 8,
    borderWidth: 1,
    borderColor: COLORS.mutedBorder,
  },
  summaryTitle: {
    fontSize: 16,
    fontWeight: '600',
    color: COLORS.textPrimary,
  },
  summaryHint: {
    fontSize: 13,
    color: COLORS.textSecondary,
  },
  errorText: {
    color: COLORS.error,
    fontSize: 14,
  },
  loadingContainer: {
    paddingVertical: 48,
    alignItems: 'center',
    justifyContent: 'center',
  },
  sections: {
    gap: 24,
  },
  section: {
    gap: 12,
  },
  sectionTitle: {
    fontSize: 18,
    fontWeight: '700',
    color: COLORS.textPrimary,
  },
  emptyMessage: {
    fontSize: 14,
    color: COLORS.textSecondary,
  },
  accountCard: {
    backgroundColor: COLORS.surface,
    borderRadius: 16,
    padding: 16,
    gap: 12,
    borderWidth: 1,
    borderColor: COLORS.mutedBorder,
  },
  accountHeader: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'flex-start',
    gap: 12,
  },
  accountTitleBlock: {
    flex: 1,
    gap: 4,
  },
  accountName: {
    fontSize: 16,
    fontWeight: '700',
    color: COLORS.textPrimary,
  },
  accountEmail: {
    fontSize: 13,
    color: COLORS.textSecondary,
  },
  accountMeta: {
    fontSize: 13,
    color: COLORS.textSecondary,
  },
  accountMetaRow: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  actionsRow: {
    flexDirection: 'row',
    gap: 12,
  },
  actionButton: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    height: 44,
    borderRadius: 12,
  },
  primaryButton: {
    backgroundColor: COLORS.primary,
  },
  primaryButtonText: {
    color: COLORS.textOnPrimary,
    fontWeight: '600',
  },
  dangerButton: {
    backgroundColor: COLORS.error,
  },
  dangerButtonText: {
    color: COLORS.textOnPrimary,
    fontWeight: '600',
  },
  statusPill: {
    paddingHorizontal: 12,
    paddingVertical: 6,
    borderRadius: 999,
  },
  statusPillText: {
    fontSize: 12,
    fontWeight: '600',
    color: COLORS.textOnPrimary,
  },
  statusPillActive: {
    backgroundColor: COLORS.success,
  },
  statusPillPending: {
    backgroundColor: COLORS.warning,
  },
  statusPillRejected: {
    backgroundColor: COLORS.error,
  },
  statusPillSuspended: {
    backgroundColor: COLORS.border,
  },
});

export default AdminDashboardScreen;
