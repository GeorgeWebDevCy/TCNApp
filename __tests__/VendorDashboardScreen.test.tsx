import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import { VendorDashboardScreen } from '../src/screens/VendorDashboardScreen';
import { translations } from '../src/localization/translations';

const resolveTranslation = (key: string): unknown => {
  return key.split('.').reduce<unknown>((current, segment) => {
    if (current && typeof current === 'object') {
      return (current as Record<string, unknown>)[segment];
    }
    return undefined;
  }, translations.en);
};

jest.mock('../src/hooks/useTransactionsAnalytics', () => ({
  useVendorTransactions: jest.fn(),
}));

jest.mock('../src/hooks/useResponsiveLayout', () => ({
  useResponsiveLayout: () => ({
    width: 768,
    height: 1024,
    shortestSide: 768,
    isLandscape: false,
    isSmallPhone: false,
    isTablet: true,
    isLargeTablet: false,
    contentPadding: 32,
    maxContentWidth: 720,
  }),
}));

jest.mock('../src/contexts/LocalizationContext', () => ({
  useLocalization: () => ({
    language: 'en',
    setLanguage: jest.fn(),
    translateError: (value: string | null) => value,
    t: (key: string, options?: { replace?: Record<string, string | number> }) => {
      const value = resolveTranslation(key);
      if (typeof value === 'string') {
        if (!options?.replace) {
          return value;
        }
        return value.replace(/{{\s*([^\s{}]+)\s*}}/g, (match, token) => {
          const replacement = options.replace?.[token];
          return replacement !== undefined ? String(replacement) : match;
        });
      }
      return key;
    },
  }),
}));

const { useVendorTransactions } = jest.requireMock(
  '../src/hooks/useTransactionsAnalytics',
);
const useVendorTransactionsMock = useVendorTransactions as jest.Mock;

describe('VendorDashboardScreen', () => {
  beforeEach(() => {
    useVendorTransactionsMock.mockReturnValue({
      transactions: [],
      monthlySavings: [
        {
          monthKey: '2024-01',
          date: new Date('2024-01-01T00:00:00.000Z'),
          totalSavings: 200,
          transactionCount: 4,
        },
      ],
      topVendors: [
        { vendorName: 'Main Branch', totalSavings: 200, transactionCount: 4 },
      ],
      statusBreakdown: [
        { status: 'pending', count: 2 },
        { status: 'completed', count: 4 },
        { status: 'failed', count: 1 },
      ],
      totalSavings: 200,
      completedCount: 4,
      isLoading: false,
      isEmpty: false,
      error: null,
      lastUpdated: new Date('2024-03-11T08:30:00.000Z'),
      refresh: jest.fn(),
    });
  });

  it('renders key vendor analytics sections', () => {
    let renderer: ReactTestRenderer.ReactTestRenderer | undefined;
    ReactTestRenderer.act(() => {
      renderer = ReactTestRenderer.create(<VendorDashboardScreen />);
    });

    const instance = renderer!.root;
    const collectText = (value: unknown): string => {
      if (typeof value === 'string' || typeof value === 'number') {
        return String(value);
      }
      if (Array.isArray(value)) {
        return value.map(collectText).join('');
      }
      return '';
    };
    const hasText = (expected: string) =>
      instance.findAll(
        node => node.type === 'Text' && collectText(node.props.children) === expected,
      ).length > 0;

    expect(hasText('Transaction analytics')).toBe(true);
    expect(hasText('Monthly savings')).toBe(true);
    expect(hasText('Top vendors')).toBe(true);
    expect(hasText('Transaction status')).toBe(true);
  });
});
