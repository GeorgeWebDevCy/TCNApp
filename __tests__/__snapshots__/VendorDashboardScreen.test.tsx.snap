// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VendorDashboardScreen renders the vendor analytics snapshot 1`] = `
<safe-area-view
  style={
    {
      "backgroundColor": "#F8FAFC",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      {
        "gap": 24,
        "padding": 24,
      }
    }
  >
    <View>
      <View
        style={
          [
            {
              "gap": 16,
              "justifyContent": "space-between",
            },
            {
              "flexDirection": "row",
            },
          ]
        }
      >
        <View
          style={
            {
              "flexShrink": 1,
              "gap": 8,
            }
          }
        >
          <Text
            style={
              {
                "color": "#0F172A",
                "fontSize": 24,
                "fontWeight": "700",
              }
            }
          >
            Transaction analytics
          </Text>
          <Text
            style={
              {
                "color": "#475569",
                "fontSize": 16,
              }
            }
          >
            Review member activity and savings at your location.
          </Text>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "gap": 12,
            }
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "backgroundColor": "#2563EB",
                "borderRadius": 12,
                "paddingHorizontal": 16,
                "paddingVertical": 10,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#FFFFFF",
                  "fontSize": 14,
                  "fontWeight": "600",
                }
              }
            >
              Refresh
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          {
            "flexDirection": "row",
            "flexWrap": "wrap",
            "gap": 16,
          }
        }
      >
        <View
          style={
            {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 16,
              "elevation": 1,
              "flex": 1,
              "gap": 8,
              "minWidth": 160,
              "padding": 16,
              "shadowColor": "#000000",
              "shadowOffset": {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.06,
              "shadowRadius": 6,
            }
          }
        >
          <Text
            style={
              {
                "color": "#475569",
                "fontSize": 14,
              }
            }
          >
            Total savings (THB)
          </Text>
          <Text
            style={
              {
                "color": "#0F172A",
                "fontSize": 22,
                "fontWeight": "700",
              }
            }
          >
            200.00
          </Text>
        </View>
        <View
          style={
            {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 16,
              "elevation": 1,
              "flex": 1,
              "gap": 8,
              "minWidth": 160,
              "padding": 16,
              "shadowColor": "#000000",
              "shadowOffset": {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.06,
              "shadowRadius": 6,
            }
          }
        >
          <Text
            style={
              {
                "color": "#475569",
                "fontSize": 14,
              }
            }
          >
            Pending transactions
          </Text>
          <Text
            style={
              {
                "color": "#0F172A",
                "fontSize": 22,
                "fontWeight": "700",
              }
            }
          >
            2
          </Text>
        </View>
      </View>
      <Text
        style={
          {
            "color": "#64748B",
            "fontSize": 12,
          }
        }
      >
        Last updated 3/11/2024, 8:30:00 AM
      </Text>
      <View
        style={
          {
            "gap": 24,
          }
        }
      >
        <View
          style={
            [
              {
                "backgroundColor": "#FFFFFF",
                "borderRadius": 16,
                "elevation": 2,
                "gap": 16,
                "padding": 16,
                "shadowColor": "#000000",
                "shadowOffset": {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.08,
                "shadowRadius": 8,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              {
                "alignItems": "flex-start",
                "flexDirection": "row",
                "gap": 12,
                "justifyContent": "space-between",
              }
            }
          >
            <View
              style={
                {
                  "flex": 1,
                }
              }
            >
              <Text
                accessibilityRole="header"
                style={
                  {
                    "color": "#0F172A",
                    "fontSize": 18,
                    "fontWeight": "600",
                  }
                }
              >
                Monthly savings
              </Text>
            </View>
            <Text
              style={
                {
                  "color": "#64748B",
                  "flexShrink": 1,
                  "fontSize": 12,
                  "textAlign": "right",
                }
              }
            >
              Total discount captured per month.
            </Text>
          </View>
          <View
            style={
              {
                "justifyContent": "center",
                "minHeight": 180,
              }
            }
          >
            <victory-chart
              domainPadding={
                {
                  "x": 24,
                  "y": 12,
                }
              }
              height={220}
              padding={
                {
                  "bottom": 48,
                  "left": 56,
                  "right": 24,
                  "top": 24,
                }
              }
              theme={{}}
            >
              <victory-axis
                style={
                  {
                    "axis": {
                      "stroke": "#CBD5E1",
                    },
                    "grid": {
                      "stroke": "transparent",
                    },
                    "tickLabels": {
                      "angle": 0,
                      "fill": "#475569",
                      "fontSize": 12,
                      "padding": 10,
                    },
                  }
                }
                tickFormat={
                  [
                    "Jan",
                  ]
                }
              />
              <victory-axis
                dependentAxis={true}
                style={
                  {
                    "axis": {
                      "stroke": "#CBD5E1",
                    },
                    "grid": {
                      "stroke": "#CBD5E1",
                      "strokeDasharray": "4 4",
                    },
                    "tickLabels": {
                      "fill": "#64748B",
                      "fontSize": 12,
                      "padding": 6,
                    },
                  }
                }
                tickFormat={[Function]}
              />
              <victory-bar
                animate={
                  {
                    "duration": 400,
                  }
                }
                data={
                  [
                    {
                      "x": "Jan",
                      "y": 200,
                    },
                  ]
                }
                style={
                  {
                    "data": {
                      "borderRadius": 6,
                      "fill": "#2563EB",
                      "width": 16,
                    },
                  }
                }
              />
            </victory-chart>
          </View>
        </View>
        <View
          style={
            [
              {
                "backgroundColor": "#FFFFFF",
                "borderRadius": 16,
                "elevation": 2,
                "gap": 16,
                "padding": 16,
                "shadowColor": "#000000",
                "shadowOffset": {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.08,
                "shadowRadius": 8,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              {
                "alignItems": "flex-start",
                "flexDirection": "row",
                "gap": 12,
                "justifyContent": "space-between",
              }
            }
          >
            <View
              style={
                {
                  "flex": 1,
                }
              }
            >
              <Text
                accessibilityRole="header"
                style={
                  {
                    "color": "#0F172A",
                    "fontSize": 18,
                    "fontWeight": "600",
                  }
                }
              >
                Top vendors
              </Text>
            </View>
            <Text
              style={
                {
                  "color": "#64748B",
                  "flexShrink": 1,
                  "fontSize": 12,
                  "textAlign": "right",
                }
              }
            >
              Where you save the most based on discount totals.
            </Text>
          </View>
          <View
            style={
              {
                "justifyContent": "center",
                "minHeight": 180,
              }
            }
          >
            <victory-chart
              domainPadding={
                {
                  "x": 16,
                  "y": 24,
                }
              }
              height={220}
              horizontal={true}
              padding={
                {
                  "bottom": 24,
                  "left": 120,
                  "right": 24,
                  "top": 24,
                }
              }
              theme={{}}
            >
              <victory-axis
                style={
                  {
                    "axis": {
                      "stroke": "#CBD5E1",
                    },
                    "grid": {
                      "stroke": "#CBD5E1",
                      "strokeDasharray": "4 4",
                    },
                    "tickLabels": {
                      "fill": "#64748B",
                      "fontSize": 12,
                      "padding": 4,
                    },
                  }
                }
              />
              <victory-axis
                dependentAxis={true}
                style={
                  {
                    "axis": {
                      "stroke": "#CBD5E1",
                    },
                    "grid": {
                      "stroke": "transparent",
                    },
                    "tickLabels": {
                      "fill": "#475569",
                      "fontSize": 12,
                    },
                  }
                }
                tickFormat={
                  [
                    "Main Branch",
                  ]
                }
              />
              <victory-bar
                animate={
                  {
                    "duration": 400,
                  }
                }
                data={
                  [
                    {
                      "x": "Main Branch",
                      "y": 200,
                    },
                  ]
                }
                style={
                  {
                    "data": {
                      "borderRadius": 6,
                      "fill": "#60A5FA",
                      "width": 16,
                    },
                  }
                }
              />
            </victory-chart>
          </View>
        </View>
        <View
          style={
            [
              {
                "backgroundColor": "#FFFFFF",
                "borderRadius": 16,
                "elevation": 2,
                "gap": 16,
                "padding": 16,
                "shadowColor": "#000000",
                "shadowOffset": {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.08,
                "shadowRadius": 8,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              {
                "alignItems": "flex-start",
                "flexDirection": "row",
                "gap": 12,
                "justifyContent": "space-between",
              }
            }
          >
            <View
              style={
                {
                  "flex": 1,
                }
              }
            >
              <Text
                accessibilityRole="header"
                style={
                  {
                    "color": "#0F172A",
                    "fontSize": 18,
                    "fontWeight": "600",
                  }
                }
              >
                Transaction status
              </Text>
            </View>
            <Text
              style={
                {
                  "color": "#64748B",
                  "flexShrink": 1,
                  "fontSize": 12,
                  "textAlign": "right",
                }
              }
            >
              Completed vs pending vs failed transactions.
            </Text>
          </View>
          <View
            style={
              {
                "justifyContent": "center",
                "minHeight": 180,
              }
            }
          >
            <View
              style={
                {
                  "flexDirection": "row",
                  "gap": 16,
                  "justifyContent": "space-between",
                }
              }
            >
              <View
                style={
                  {
                    "alignItems": "center",
                    "height": 180,
                    "justifyContent": "center",
                    "width": 180,
                  }
                }
              >
                <victory-pie
                  animate={
                    {
                      "duration": 400,
                    }
                  }
                  colorScale={
                    [
                      "#2563EB",
                      "#F59E0B",
                      "#DC2626",
                    ]
                  }
                  data={
                    [
                      {
                        "x": "Completed",
                        "y": 4,
                      },
                      {
                        "x": "Pending",
                        "y": 2,
                      },
                      {
                        "x": "Failed",
                        "y": 1,
                      },
                    ]
                  }
                  innerRadius={60}
                  labels={[Function]}
                />
                <View
                  style={
                    {
                      "alignItems": "center",
                      "justifyContent": "center",
                      "position": "absolute",
                    }
                  }
                >
                  <Text
                    style={
                      {
                        "color": "#0F172A",
                        "fontSize": 16,
                        "fontWeight": "600",
                        "textAlign": "center",
                      }
                    }
                  >
                    4 completed
                  </Text>
                </View>
              </View>
              <View
                style={
                  {
                    "flex": 1,
                    "gap": 12,
                    "justifyContent": "center",
                  }
                }
              >
                <View
                  style={
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "gap": 8,
                    }
                  }
                >
                  <View
                    style={
                      [
                        {
                          "borderRadius": 6,
                          "height": 12,
                          "width": 12,
                        },
                        {
                          "backgroundColor": "#2563EB",
                        },
                      ]
                    }
                  />
                  <Text
                    style={
                      {
                        "color": "#475569",
                        "fontSize": 14,
                      }
                    }
                  >
                    Completed
                     · 
                    4
                  </Text>
                </View>
                <View
                  style={
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "gap": 8,
                    }
                  }
                >
                  <View
                    style={
                      [
                        {
                          "borderRadius": 6,
                          "height": 12,
                          "width": 12,
                        },
                        {
                          "backgroundColor": "#F59E0B",
                        },
                      ]
                    }
                  />
                  <Text
                    style={
                      {
                        "color": "#475569",
                        "fontSize": 14,
                      }
                    }
                  >
                    Pending
                     · 
                    2
                  </Text>
                </View>
                <View
                  style={
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "gap": 8,
                    }
                  }
                >
                  <View
                    style={
                      [
                        {
                          "borderRadius": 6,
                          "height": 12,
                          "width": 12,
                        },
                        {
                          "backgroundColor": "#DC2626",
                        },
                      ]
                    }
                  />
                  <Text
                    style={
                      {
                        "color": "#475569",
                        "fontSize": 14,
                      }
                    }
                  >
                    Failed
                     · 
                    1
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</safe-area-view>
`;
